src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGANaivelySpark.java

```diff
@@ -0,0 +1,485 @@
+
+    /**
+     * Performs assembly on the FASTA files. Essentially a wrapper around a limited number of modules in SGA.
+     * @param fastqFilesForEachBreakPoint temporary FASTA file of sequences around the putative breakpoint
+     * @param sgaPath                     full path to SGA
+     * @param threads                     number of threads to use in various modules, if the module supports parallelism
+     * @param runCorrections              user's decision to run sga's corrections (with default parameter values) or not
+     * @return                            contig file and runtime information, associated with the breakpoint id
+     */
+    private static Tuple2<Long, SGAAssemblyResultType> performAssembly(final Tuple2<Long, File> fastqFilesForEachBreakPoint,
+                                                                       final Path sgaPath,
+                                                                       final int threads,
+                                                                       final boolean runCorrections){
+
+        final File rawFASTQ = fastqFilesForEachBreakPoint._2();
+
+        final SGAAssemblyResultType assembledContigsFileAndRuntimeInfo = SGASerialRunner(sgaPath, rawFASTQ, threads, runCorrections);
+
+        return new Tuple2<>(fastqFilesForEachBreakPoint._1(), assembledContigsFileAndRuntimeInfo);
+    }
+
+    @VisibleForTesting
+    static SGAAssemblyResultType SGASerialRunner(final Path sgaPath, final File rawFASTQ, final int threads, final boolean runCorrections){
+
+        int threadsToUse = threads;
+        if( System.getProperty("os.name").toLowerCase().contains("mac") && threads>1){
+            System.err.println("Running on Mac OS X, which doesn't provide unnamed semaphores used by SGA. " +
+                               "Resetting threads argument to 1. ");
+            threadsToUse = 1;
+        }
+        // store every intermediate file in the same temp directory as the raw fastq file
+        final File workingDir = rawFASTQ.getParentFile();
+
+        // the index module is used frequently, so make single instance and pass around
+        final SGAModule indexer = new SGAModule("index");
+        final ArrayList<String> indexerArgs = new ArrayList<>();
+        indexerArgs.add("--algorithm"); indexerArgs.add("ropebwt");
+        indexerArgs.add("--threads");   indexerArgs.add(Integer.toString(threadsToUse));
+        indexerArgs.add("--check");
+        indexerArgs.add("");
+
+        // collect runtime information along the way
+        final List<Tuple2<String, Tuple3<Integer, String, String>>> runtimeInfo = new ArrayList<>();
+
+        String preppedFileName = runAndStopEarly("preprocess", rawFASTQ, sgaPath, workingDir, threadsToUse, indexer, indexerArgs, runtimeInfo);
+        if( null == preppedFileName ){
+            return new SGAAssemblyResultType(null, runtimeInfo);
+        }
+
+        final File preprocessedFile = new File(workingDir, preppedFileName);
```

Comment:
This isn't used except inside the corrections if() block, might was well move it in there.